runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -1030,27 +1026,22 @@ func isTimeRangeNil(tr *runtimev1.TimeRange) bool {
 	return tr == nil || (tr.Start == nil && tr.End == nil)
 }
 
-func aliasesToMeasureMap(aliases []*runtimev1.MetricsViewComparisonMeasureAlias, measureAliases map[string]columnIdentifier, hasComparison bool) error {
+func validateMeasureAliases(aliases []*runtimev1.MetricsViewComparisonMeasureAlias, measures []*runtimev1.MetricsViewAggregationMeasure, hasComparison bool) error {
 	for _, alias := range aliases {
+		// We need to make sure the aliases measure is selected. Having query will fail otherwise
+		if !slices.ContainsFunc(measures, func(measure *runtimev1.MetricsViewAggregationMeasure) bool {
+			return measure.Name == alias.Name
+		}) {
+			return fmt.Errorf("measure alias not selected: %s", alias.Name)
+		}
```

Comment:
Maybe redundant? IMO it's okay to pass an alias that's not used in the expression. And fine to just fail at query time if using an expression that's not selected.